x/evm/keeper/msg_server.go

```diff
@@ -592,10 +592,10 @@ func (k Keeper) convertCoinNativeERC20(
 
 	recipientBalanceBefore, err := k.ERC20().BalanceOf(erc20Addr, recipient, ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to retrieve balance")
+		return nil, errors.Wrap(err, "failed to retrieve recipient balance")
 	}
 	if recipientBalanceBefore == nil {
-		return nil, fmt.Errorf("failed to retrieve balance, balance is nil")
+		return nil, fmt.Errorf("failed to retrieve recipient balance, balance is nil")
```

Comment:
Since we only intend for ERC20 transfers to properly send an amount that alters the recipient balance, let's put the `balanceOf` recipient checks before and after to inside of `ERC20().Transfer`. That'll make using it more safe.